src/bidiMapper/domains/context/BrowsingContextImpl.ts

```diff
@@ -576,6 +569,21 @@ export class BrowsingContextImpl {
     this.#resetLifecycleIfFinished();
 
     this.#loaderId = loaderId;
+    if (this.#loaderId) {
+      this.#eventManager.registerEvent(
+        {
+          type: 'event',
+          method: ChromiumBidi.BrowsingContext.EventNames.NavigationStarted,
```

Comment:
The PR title is misleading. `#documentChanged` is called after CDP call `Page.navigate` or after `Page.lifecycleEvent: init`, and has nothing to do with `requestWillBeSent`.